Error building mozjs_sys

Hi. I'm running the build using the command `./mach build --release` and all of the crates install but one. I've deleted the `Cargo.lock` file and then restarted on the off chance that the file was corrupted on download (maybe?), but the error is still happening. The error says that it's looking for a file and it can't find it. I saw another discussion that the ordering of the crates may be off and it's looking for a dependency that hasn't been downloaded yet. Just a heads up. I want to take a look at Servo, but I can't run it. 

```
error: failed to run custom build command for `mozjs_sys v0.68.2 (https://github.com/servo/mozjs#0623d5fc)`

Caused by:
  process didn't exit successfully: `/Users/peterweyand/Code/servo/target/release/build/mozjs_sys-07fc63b58f2bfa29/build-script-build` (exit status: 101)
  --- stdout
  cargo:outdir=/Users/peterweyand/Code/servo/target/release/build/mozjs_sys-152435a8df71f11e/out/build
  [[ '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/configure -ot '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/configure.in ]] && touch '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/configure || true
  [[ '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/old-configure -ot '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/old-configure.in ]] && touch '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/old-configure || true
  ! [[ '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/configure.in -ot '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/configure ]] && touch '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/configure || true
  ! [[ '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/old-configure.in -ot '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/old-configure ]] && touch '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/old-configure || true
  if [[ '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/configure -nt config.status ]] ; then \
  	  PYTHON="python2" MOZ_TOOLS="" \
  	  CC="clang" CFLAGS="" \
  	  CPP="gcc -E" CPPFLAGS="-I/Users/peterweyand/.cargo/registry/src/github.com-1ecc6299db9ec823/encoding_c_mem-0.2.6/include " \
  	  CXX="clang++" CXXFLAGS="-stdlib=libc++" \
  	  AS="" AR="ar" \
  	  STLPORT_LIBS="" \
  	  '/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs'/js/src/configure --disable-jemalloc --disable-js-shell --disable-tests --disable-export-js --disable-shared-js --build-backends=RecursiveMake --enable-js-streams || (cat config.log && exit 1) ; \
  	fi
  created virtual environment CPython3.10.8.final.0-64 in 50ms
    creator CPython3Posix(dest=/Users/peterweyand/Code/servo/target/release/build/mozjs_sys-152435a8df71f11e/out/build/_virtualenvs/build, clear=False, no_vcs_ignore=False, global=False)
    activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator
  Using Python 3.10.8 from /Users/peterweyand/Code/servo/python/_virtualenv3.10/bin/python3
  Site not up-to-date reason: "/Users/peterweyand/Code/servo/target/release/build/mozjs_sys-152435a8df71f11e/out/build/_virtualenvs/build" does not exist

  --- stderr
  Traceback (most recent call last):
    File "/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs/js/src/../../configure.py", line 350, in <module>
      sys.exit(main(sys.argv))
    File "/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs/js/src/../../configure.py", line 121, in main
      _activate_build_virtualenv()
    File "/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs/js/src/../../configure.py", line 344, in _activate_build_virtualenv
      if not build_site.ensure():
    File "/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs/python/mach/mach/site.py", line 565, in ensure
      _create_venv_with_pthfile(
    File "/Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/mozjs/python/mach/mach/site.py", line 1191, in _create_venv_with_pthfile
      with open(os.path.join(platlib_site_packages_dir, PTH_FILENAME), "w") as f:
  FileNotFoundError: [Errno 2] No such file or directory: '/Users/peterweyand/Code/servo/target/release/build/mozjs_sys-152435a8df71f11e/out/build/_virtualenvs/build/lib/python3.10/site-packages/mach.pth'
  cat: config.log: No such file or directory
  make: *** [maybe-configure] Error 1
  thread 'main' panicked at 'assertion failed: result.success()', /Users/peterweyand/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/0623d5f/mozjs/build.rs:178:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
      Timing report saved to /Users/peterweyand/Code/servo/target/cargo-timings/cargo-timing-20221215T003457Z.html
[Warning] Could not generate notification! 
Build FAILED in 0:02:27
```

I'm running into this issue on macOS Monterey as well. I can build mozjs separately from a git clone using both the native python 3.9 and Homebrew's python 3.9, but I building this crate under servo's `mach build` fails consistently.

Here's output from using Monterey's native python 3.9 (3.9.6, I think) and the command line tools (not Xcode):

```
~/git/servo (git)-[master] % ./mach -v build --release
info: component 'llvm-tools' for target 'x86_64-apple-darwin' is up to date
    Updating git repository `https://github.com/servo/surfman-chains.git`
    Updating git repository `https://github.com/jrmuizel/raqote`
  Downloaded lyon_geom v1.0.4
  Downloaded lyon_geom v0.17.7
  Downloaded libm v0.2.6
  Downloaded sw-composite v0.7.15
  Downloaded euclid v0.22.7
  Downloaded surfman v0.5.0
  Downloaded plane-split v0.17.1
  Downloaded 7 crates (447.9 KB) in 0.94s
   Compiling libc v0.2.121

[... snip ...]

   Compiling raqote v0.8.2 (https://github.com/jrmuizel/raqote#7b235e9a)
   Compiling compositing v0.0.1 (/Users/jparnzen/git/servo/components/compositing)
error: failed to run custom build command for `mozjs_sys v0.68.2 (https://github.com/servo/mozjs#f87705d6)`

Caused by:
  process didn't exit successfully: `/Users/jparnzen/git/servo/target/release/build/mozjs_sys-f4273b46230a01ce/build-script-build` (exit status: 101)
  --- stdout
  cargo:outdir=/Users/jparnzen/git/servo/target/release/build/mozjs_sys-310a5510b9aece22/out/build
  [[ '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure -ot '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure.in ]] && touch '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure || true
  [[ '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure -ot '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure.in ]] && touch '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure || true
  ! [[ '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure.in -ot '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure ]] && touch '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure || true
  ! [[ '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure.in -ot '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure ]] && touch '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/old-configure || true
  if [[ '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure -nt config.status ]] ; then \
  	  PYTHON="python2" MOZ_TOOLS="" \
  	  CC="clang" CFLAGS="" \
  	  CPP="gcc -E" CPPFLAGS="-I/Users/jparnzen/.cargo/registry/src/github.com-1ecc6299db9ec823/encoding_c_mem-0.2.6/include " \
  	  CXX="clang++" CXXFLAGS="-stdlib=libc++" \
  	  AS="" AR="ar" \
  	  STLPORT_LIBS="" \
  	  '/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs'/js/src/configure --disable-jemalloc --disable-js-shell --disable-tests --disable-export-js --disable-shared-js --build-backends=RecursiveMake --enable-js-streams || (cat config.log && exit 1) ; \
  	fi
  created virtual environment CPython3.9.6.final.0-64 in 103ms
    creator CPython3macOsFramework(dest=/Users/jparnzen/git/servo/target/release/build/mozjs_sys-310a5510b9aece22/out/build/_virtualenvs/build, clear=False, no_vcs_ignore=False, global=False)
    activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator
  Using Python 3.9.6 from /Users/jparnzen/git/servo/python/_virtualenv3.9/bin/python3
  Site not up-to-date reason: "/Users/jparnzen/git/servo/target/release/build/mozjs_sys-310a5510b9aece22/out/build/_virtualenvs/build" does not exist
  Created Python 3 virtualenv
  checking for vcs source checkout... no
  checking for a shell... /bin/sh
  checking for host system type... x86_64-apple-darwin21.6.0
  checking for target system type... x86_64-apple-darwin21.6.0
  checking whether cross compiling... no
  checking for Python 3... /Users/jparnzen/git/servo/target/release/build/mozjs_sys-310a5510b9aece22/out/build/_virtualenvs/build/bin/python (3.9.6)
  checking for wget... /usr/local/bin/wget
  checking for ccache... not found
  Using target sysroot in /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk
  Using host sysroot in /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk
  checking for the target C compiler... /usr/bin/clang
  checking whether the target C compiler can be used... yes
  checking the target C compiler version... 12.0.0.or.more
  checking the target C compiler works... yes
  checking for the target C++ compiler... /usr/bin/clang++
  checking whether the target C++ compiler can be used... yes
  checking the target C++ compiler version... 12.0.0.or.more
  checking the target C++ compiler works... yes
  checking for the host C compiler... /usr/bin/clang
  checking whether the host C compiler can be used... yes
  checking the host C compiler version... 12.0.0.or.more
  checking the host C compiler works... yes
  checking for the host C++ compiler... /usr/bin/clang++
  checking whether the host C++ compiler can be used... yes
  checking the host C++ compiler version... 12.0.0.or.more
  checking the host C++ compiler works... yes
  checking for linker... ld64
  checking for 64-bit OS... yes
  checking for the assembler... /usr/bin/clang
  checking for ar... /usr/bin/ar
  checking for strndup... yes
  checking for posix_memalign... yes
  checking for memalign... no
  checking for malloc_usable_size... no
  checking for stdint.h... yes
  checking for inttypes.h... yes
  checking for malloc.h... no
  checking for alloca.h... yes
  checking for sys/byteorder.h... no
  checking for getopt.h... yes
  checking for unistd.h... yes
  checking for nl_types.h... yes
  checking for cpuid.h... yes
  checking for fts.h... yes
  checking for sys/statvfs.h... yes
  checking for sys/statfs.h... no
  checking for sys/vfs.h... no
  checking for sys/mount.h... yes
  checking for sys/quota.h... no
  checking for sys/queue.h... yes
  checking for sys/types.h... yes
  checking for netinet/in.h... yes
  checking for byteswap.h... no
  checking for memfd_create in sys/mman.h... no
  checking for perf_event_open system call... no
  checking whether the C compiler supports -Wbitfield-enum-conversion... yes
  checking whether the C++ compiler supports -Wbitfield-enum-conversion... yes
  checking whether the C++ compiler supports -Wdeprecated-this-capture... yes
  checking whether the C compiler supports -Wformat-type-confusion... yes
  checking whether the C++ compiler supports -Wformat-type-confusion... yes
  checking whether the C compiler supports -Wshadow-field-in-constructor-modified... yes
  checking whether the C++ compiler supports -Wshadow-field-in-constructor-modified... yes
  checking whether the C compiler supports -Wno-error=tautological-type-limit-compare... yes
  checking whether the C++ compiler supports -Wno-error=tautological-type-limit-compare... yes
  checking whether the C compiler supports -Wunreachable-code-return... yes
  checking whether the C++ compiler supports -Wunreachable-code-return... yes
  checking whether the C compiler supports -Wunused-but-set-parameter... yes
  checking whether the C++ compiler supports -Wunused-but-set-parameter... yes
  checking whether the C compiler supports -Wclass-varargs... yes
  checking whether the C++ compiler supports -Wclass-varargs... yes
  checking whether the C++ compiler supports -Wempty-init-stmt... yes
  checking whether the C compiler supports -Wfloat-overflow-conversion... yes
  checking whether the C++ compiler supports -Wfloat-overflow-conversion... yes
  checking whether the C compiler supports -Wfloat-zero-conversion... yes
  checking whether the C++ compiler supports -Wfloat-zero-conversion... yes
  checking whether the C compiler supports -Wloop-analysis... yes
  checking whether the C++ compiler supports -Wloop-analysis... yes
  checking whether the C compiler supports -Wno-range-loop-analysis... yes
  checking whether the C++ compiler supports -Wno-range-loop-analysis... yes
  checking whether the C++ compiler supports -Wc++2a-compat... yes
  checking whether the C++ compiler supports -Wcomma-subscript... no
  checking whether the C compiler supports -Wenum-compare-conditional... yes
  checking whether the C++ compiler supports -Wenum-compare-conditional... yes
  checking whether the C++ compiler supports -Wno-ambiguous-reversed-operator... yes
  checking whether the C++ compiler supports -Wno-error=deprecated... yes
  checking whether the C++ compiler supports -Wno-error=deprecated-anon-enum-enum-conversion... yes
  checking whether the C++ compiler supports -Wno-error=deprecated-enum-enum-conversion... yes
  checking whether the C++ compiler supports -Wno-error=deprecated-enum-float-conversion... yes
  checking whether the C++ compiler supports -Wno-error=deprecated-pragma... yes
  checking whether the C++ compiler supports -Wno-error=deprecated-this-capture... yes
  checking whether the C++ compiler supports -Wno-error=deprecated-volatile... yes
  checking whether the C++ compiler supports -Wno-error=volatile... no
  checking whether the C++ compiler supports -Wcomma... yes
  checking whether the C compiler supports -Wduplicated-cond... no
  checking whether the C++ compiler supports -Wduplicated-cond... no
  checking whether the C++ compiler supports -Wimplicit-fallthrough... yes
  checking whether the C compiler supports -Wlogical-op... no
  checking whether the C++ compiler supports -Wlogical-op... no
  checking whether the C compiler supports -Wduplicate-method-arg... yes
  checking whether the C++ compiler supports -Wduplicate-method-arg... yes
  checking whether the C compiler supports -Wduplicate-method-match... yes
  checking whether the C++ compiler supports -Wduplicate-method-match... yes
  checking whether the C compiler supports -Wmissing-method-return-type... yes
  checking whether the C++ compiler supports -Wmissing-method-return-type... yes
  checking whether the C compiler supports -Wobjc-signed-char-bool... yes
  checking whether the C++ compiler supports -Wobjc-signed-char-bool... yes
  checking whether the C compiler supports -Wsemicolon-before-method-body... yes
  checking whether the C++ compiler supports -Wsemicolon-before-method-body... yes
  checking whether the C compiler supports -Wsuper-class-method-mismatch... yes
  checking whether the C++ compiler supports -Wsuper-class-method-mismatch... yes
  checking whether the C compiler supports -Wstring-conversion... yes
  checking whether the C++ compiler supports -Wstring-conversion... yes
  checking whether the C++ compiler supports -Wno-inline-new-delete... yes
  checking whether the C compiler supports -Wno-error=maybe-uninitialized... no
  checking whether the C++ compiler supports -Wno-error=maybe-uninitialized... no
  checking whether the C compiler supports -Wno-error=deprecated-declarations... yes
  checking whether the C++ compiler supports -Wno-error=deprecated-declarations... yes
  checking whether the C compiler supports -Wno-error=array-bounds... yes
  checking whether the C++ compiler supports -Wno-error=array-bounds... yes
  checking whether the C compiler supports -Wno-error=free-nonheap-object... yes
  checking whether the C++ compiler supports -Wno-error=free-nonheap-object... yes
  checking whether the C compiler supports -Wno-multistatement-macros... no
  checking whether the C++ compiler supports -Wno-multistatement-macros... no
  checking whether the C compiler supports -Wno-error=class-memaccess... no
  checking whether the C++ compiler supports -Wno-error=class-memaccess... no
  checking whether the C compiler supports -Wno-error=atomic-alignment... yes
  checking whether the C++ compiler supports -Wno-error=atomic-alignment... yes
  checking whether the C compiler supports -Wno-error=deprecated-builtins... no
  checking whether the C++ compiler supports -Wno-error=deprecated-builtins... no
  checking whether the C compiler supports -Wformat... yes
  checking whether the C++ compiler supports -Wformat... yes
  checking whether the C compiler supports -Wformat-security... yes
  checking whether the C++ compiler supports -Wformat-security... yes
  checking whether the C compiler supports -Wformat-overflow=2... no
  checking whether the C++ compiler supports -Wformat-overflow=2... no
  checking whether the C compiler supports -Werror=implicit-function-declaration... yes
  checking whether the C compiler supports -Wno-psabi... yes
  checking whether the C++ compiler supports -Wno-psabi... yes
  checking whether the C compiler supports -Wthread-safety... yes
  checking whether the C++ compiler supports -Wthread-safety... yes
  checking whether the C compiler supports -Werror=unguarded-availability-new... yes
  checking whether the C++ compiler supports -Werror=unguarded-availability-new... yes
  checking whether the C++ compiler supports -fno-sized-deallocation... yes
  checking whether the C++ compiler supports -fno-aligned-new... yes
  checking for llvm_profdata... not found
  checking for rustfmt... /Users/jparnzen/.cargo/bin/rustfmt
  checking for tm_zone and tm_gmtoff in struct tm... yes
  checking for getpagesize... yes
  checking for gmtime_r... yes
  checking for localtime_r... yes
  checking for gettid... no
  checking for setpriority... yes
  checking for syscall... yes
  checking for getc_unlocked... yes
  checking for pthread_getname_np... yes
  checking for pthread_get_name_np... no
  checking for strerror... yes
  checking for __cxa_demangle... yes
  checking for unwind.h... yes
  checking for _Unwind_Backtrace... yes
  checking for _getc_nolock... no
  checking for localeconv... yes
  checking for awk... /usr/local/bin/awk
  checking for perl... /usr/local/bin/perl
  checking for minimum required perl version >= 5.006... 5.036000
  checking for full perl installation... yes
  checking for gmake... /usr/local/bin/gmake
  checking for watchman... not found
  checking for xargs... /usr/bin/xargs
  checking for dsymutil... /usr/bin/dsymutil
  checking for mkfshfs... /sbin/newfs_hfs
  checking for hfs_tool... not found
  checking for llvm-objdump... /Library/Developer/CommandLineTools/usr/bin/llvm-objdump
  checking for strip... /usr/bin/strip
  checking for m4... /usr/bin/gm4
  creating cache ./config.cache
  checking host system type... x86_64-apple-darwin21.6.0
  checking target system type... x86_64-apple-darwin21.6.0
  checking build system type... x86_64-apple-darwin21.6.0
  checking for Cygwin environment... no
  checking for mingw32 environment... no
  checking for executable suffix... no
  checking for gcc... (cached) /usr/bin/clang -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -mmacosx-version-min=10.12 -std=gnu99
  checking whether the C compiler (/usr/bin/clang -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -mmacosx-version-min=10.12 -std=gnu99   ) works... (cached) yes
  checking whether the C compiler (/usr/bin/clang -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -mmacosx-version-min=10.12 -std=gnu99   ) is a cross-compiler... no
  checking whether we are using GNU C... (cached) yes
  checking whether /usr/bin/clang -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -mmacosx-version-min=10.12 -std=gnu99 accepts -g... (cached) yes
  checking for c++... (cached) /usr/bin/clang++ -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -mmacosx-version-min=10.12 -stdlib=libc++ -std=gnu++17
  checking whether the C++ compiler (/usr/bin/clang++ -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -mmacosx-version-min=10.12 -stdlib=libc++ -std=gnu++17 -stdlib=libc++  ) works... (cached) yes
  checking whether the C++ compiler (/usr/bin/clang++ -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -mmacosx-version-min=10.12 -stdlib=libc++ -std=gnu++17 -stdlib=libc++  ) is a cross-compiler... no
  checking whether we are using GNU C++... (cached) yes
  checking whether /usr/bin/clang++ -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -mmacosx-version-min=10.12 -stdlib=libc++ -std=gnu++17 accepts -g... (cached) yes
  checking for --noexecstack option to as... yes
  checking for -z noexecstack option to ld... no
  checking for -z text option to ld... no
  checking for -z relro option to ld... no
  checking for -z nocopyreloc option to ld... no
  checking for -Bsymbolic-functions option to ld... no
  checking for --build-id=sha1 option to ld... no
  checking for -dead_strip option to ld... yes
  checking for working const... yes
  checking for mode_t... yes
  checking for off_t... yes
  checking for pid_t... yes
  checking for size_t... yes
  checking for ssize_t... yes
  checking whether 64-bits std::atomic requires -latomic... no
  checking for dirent.h that defines DIR... yes
  checking for opendir in -ldir... no
  checking for gethostbyname_r in -lc_r... no
  checking for dladdr... yes
  checking for socket in -lsocket... no
  checking whether /usr/bin/clang -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -mmacosx-version-min=10.12 -std=gnu99 accepts -pthread... yes
  checking for clock_gettime(CLOCK_MONOTONIC)... no
  checking for sin in -lm... yes
  checking for res_ninit()... no
  checking for nl_langinfo and CODESET... yes
  checking for __thread keyword for TLS variables... yes
  checking for valid optimization flags... yes
  checking for -pipe support... yes
  checking what kind of list files are supported by the linker... filelist
  checking for malloc.h... no
  checking for malloc_np.h... no
  checking for malloc/malloc.h... yes
  checking for valloc in malloc.h... no
  checking for valloc in unistd.h... yes
  checking for _aligned_malloc in malloc.h... no
  updating cache ./config.cache
  creating ./config.data
  Creating config.status
   0:00.89 File already read. Skipping: /Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs/build/unix/moz.build
  /Library/Developer/CommandLineTools/usr/bin/make -f Makefile
  /Library/Developer/CommandLineTools/usr/bin/make recurse_pre-export
  /Users/jparnzen/git/servo/target/release/build/mozjs_sys-310a5510b9aece22/out/build/_virtualenvs/build/bin/python -m mozbuild.action.process_install_manifest --track install_dist_include.track dist/include _build_manifests/install/dist_include
  Elapsed: 0.30s; From dist/include: Kept 3 existing; Added/updated 535; Removed 0 files and 0 directories.
  BUILDTASK {"argv": ["/Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/mozjs/python/mozbuild/mozbuild/action/process_install_manifest.py", "--track", "install_dist_include.track", "dist/include", "_build_manifests/install/dist_include"], "start": 1675918361.6542132, "end": 1675918361.9516191, "context": null}

  --- stderr
  gmake: *** No targets specified and no makefile found.  Stop.
  Reticulating splines...
  Finished reading 62 moz.build files in 0.60s
  Read 0 gyp files in parallel contributing 0.00s to total wall time
  Processed into 253 build config descriptors in 0.71s
  RecursiveMake backend executed in 1.09s
    174 total backend files; 174 created; 0 updated; 0 unchanged; 0 deleted; 6 -> 39 Makefile
  Total wall time: 2.52s; CPU time: 1.07s; Efficiency: 42%; Untracked: 0.12s
  make[3]: *** read jobs pipe: Resource temporarily unavailable.  Stop.
  make[3]: *** Waiting for unfinished jobs....
  make[2]: *** [pre-export] Error 2
  make[1]: *** [default] Error 2
  make: *** [all] Error 2
  thread 'main' panicked at 'assertion failed: result.success()', /Users/jparnzen/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f87705d/mozjs/build.rs:178:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
      Timing report saved to /Users/jparnzen/git/servo/target/cargo-timings/cargo-timing-20230209T044724Z.html
[Warning] Could not generate notification! 
Build FAILED in 0:06:18
```

Using Homebrew's python 3.9 causes different output but I think it's from the same cause: some sort of missing build target or missing makefile.

I've deleted and re-cloned the repo,  but I didn't try clearing the cargo cache. Any ideas for how to get this building?

--- a/.github/workflows/mac.yml
+++ b/.github/workflows/mac.yml
@@ -54,15 +54,6 @@ jobs:
           fetch-depth: 2
       - name: Run sccache-cache
         uses: mozilla-actions/sccache-action@v0.0.3
-      # TODO: Remove this step when the compatibility issue between mozjs and
-      #       Homebrew's Python 3.10 formula (servo/rust-mozjs#559) is fixed
-      - name: Select Python 3.9
-        run: |
-          brew install python@3.9
-          cd $(dirname $(which python3.9))
-          rm -f python3 pip3
-          ln -s python3.9 python3
-          ln -s pip3.9 pip3
       - name: Bootstrap
         run: |
           python3 -m pip install --upgrade pip virtualenv

